docs/CONTROLFLOW.md

```diff
@@ -0,0 +1,343 @@
+# Control Flow Obfuscation
+
+> **This feature is experimental**. To enable it, set the environment variable `GARBLE_EXPERIMENTAL_CONTROLFLOW=1`
+
+### Mechanism
+
+
+Control flow obfuscation works in several stages:
+
+1) Collect functions with `//garble:controlflow` comment
+2) Converts [go/ast](https://pkg.go.dev/go/ast) representation to [go/ssa](https://pkg.go.dev/golang.org/x/tools/go/ssa)
+3) Applies [block splitting](#block-splitting)
+4) Generates [junk jumps](#junk-jumps)
+5) Applies [control flow flattening](#control-flow-flattening)
+6) Converts go/ssa back into go/ast
+
+### Example usage
+
+```go
+
+// Obfuscate with defaults parameters
+//garble:controlflow
+func main() {
+	println("Hello world!")
+}
+```
+
+### Parameter explanation
+
+> Unlike other garble features (which just work), we recommend that you understand how parameters affect control flow obfuscation and which caveats exist.
+
+> Code snippets below without name obfuscation, for better readability.
+
+#### Block splitting
+
+Param: `block_splits` (default: `0`)
```

Comment:
Say "parameter" instead of "param" throughout the doc; no need to abbreviate in a markdown file.